Assets/HoloToolkit/SpatialUnderstanding/Scripts/SpatialUnderstanding.cs

```diff
@@ -0,0 +1,244 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using UnityEngine;
+using System.Collections;
+using HoloToolkit.Unity;
+using System.Collections.Generic;
+using System.Runtime.InteropServices;
+using System;
+
+namespace HoloToolkit.Unity
+{
+    /// <summary>
+    /// The SpatialUnderstanding class controls the state and flow of the 
+    /// scanning process used in the understanding module. 
+    /// </summary>
+    [RequireComponent(typeof(SpatialUnderstandingSourceMesh))]
+    [RequireComponent(typeof(SpatialUnderstandingCustomMesh))]
+    public class SpatialUnderstanding : Singleton<SpatialUnderstanding>
+    {
+        // Consts
+        public const float ScanSearchDistance = 8.0f;
+
+        // Enums
+        public enum ScanStates
+        {
+            None,
+            ReadyToScan,
+            Scanning,
+            Finishing,
+            Done
+        }
+
+        // Config
+        [Tooltip("If set to false, scanning will only begin after RequestBeginScanning is called")]
+        public bool AutoBeginScanning = true;
+        [Tooltip("Update period used during the scanning process (typically faster than after scanning is completed)")]
+        public float UpdatePeriod_DuringScanning = 1.0f;
```

Comment:
I'm not sure if this is nit picking or not (I don't personally think it is) but these member variable names don't match the naming conventions for the rest of the toolkit. I've seen both ProperCase and camelCase but I have not seen any use of the underscore in inspector variable names. Great use of the tooltip though. Thank you for that.